Firrtl exception when using firrtl.transforms.Flatten.

<!--
This repository's issues are reserved for feature requests and bug reports.
Please select the item best describing the issue in each category and delete the other items.
-->
<!-- choose all that apply -->
**Type of issue**: bug report 

  * **What is the current behavior?**

Using the `firrtl.transforms.Flatten` can sometimes cause the following firrtl exception:

```
Exception in thread "main" firrtl.FIRRTLException: No EmittedCircuit found! Did you delete any annotations?
```

@jackkoenig was able to reproduce the error using Flatten as follows:
```
java -cp ./utils/bin/firrtl.jar firrtl.Driver -i regress/FPU.fir -o out -X verilog -fct firrtl.transforms.Flatten
```
As for myself, I used InlineInstance in a couple of places, and noticed if I set the custom-transform pass to include Flatten, I also run into this exception:

I have added `firrtl.Driver --custom-transforms=firrtl.passes.wiring.WiringTransform,firrtl.passes.InlineInstances,firrtl.transforms.Flatten` 

In my case, I am using rocket's `trait InlineInstance` to invoke the InlineAnnotation:

```scala
trait InlineInstance { self: chisel3.experimental.BaseModule =>
  chisel3.experimental.annotate(
    new chisel3.experimental.ChiselAnnotation {
      def toFirrtl: firrtl.annotations.Annotation = firrtl.passes.InlineAnnotation(self.toNamed) } )
}
```

  * **Please tell us about your environment:**
    - firrtl version: 50ba95fe05cc348496425733554be5fc7d797de1
    - chisel3 version: https://github.com/freechipsproject/chisel3/commit/9a0ce2272c9d5d0a8bdc90e84269749ce054664d

**The error:** 

```
Total FIRRTL Compile Time: 184664.0 ms
Exception in thread "main" firrtl.FIRRTLException: No EmittedCircuit found! Did you delete any annotations?
... [snip] ...
	at firrtl.CircuitState.getEmittedCircuit(Compiler.scala:48)
	at firrtl.Driver$.$anonfun$execute$1(Driver.scala:260)
	at logger.Logger$.$anonfun$makeScope$2(Logger.scala:138)
	at scala.util.DynamicVariable.withValue(DynamicVariable.scala:58)
	at logger.Logger$.makeScope(Logger.scala:136)
	at firrtl.Driver$.execute(Driver.scala:224)
	at firrtl.Driver$.execute(Driver.scala:302)
	at firrtl.Driver$.main(Driver.scala:318)
	at firrtl.Driver.main(Driver.scala)
make: *** [snip] Error 1

```

--- a/src/main/scala/firrtl/transforms/Flatten.scala
+++ b/src/main/scala/firrtl/transforms/Flatten.scala
@@ -104,7 +104,7 @@ class Flatten extends Transform {
    override def execute(state: CircuitState): CircuitState = {
      val annos = state.annotations.collect { case a @ FlattenAnnotation(_) => a }
      annos match {
-       case Nil => CircuitState(state.circuit, state.form)
+       case Nil => state
        case myAnnotations =>
          val c = state.circuit
          val (modNames, instNames) = collectAnns(state.circuit, myAnnotations)

--- a/src/test/scala/firrtlTests/FlattenTests.scala
+++ b/src/test/scala/firrtlTests/FlattenTests.scala
@@ -224,4 +224,14 @@ class FlattenTests extends LowTransformSpec {
           |    b <= a""".stripMargin
      execute(input, check, Seq(flatten("Inline1")))
   }
+  "The Flatten transform" should "do nothing if no flatten annotations are present" in{
+    val input =
+      """|circuit Foo:
+         |  module Foo:
+         |    input a: UInt<1>
+         |    output b: UInt<1>
+         |    b <= a
+         |""".stripMargin
+    execute(input, input, Seq.empty)
+  }
 }

